server/uploadHandler/src/createNewJob.ts

```diff
@@ -1,79 +1,86 @@
 import { v4 as uuidv4 } from "uuid"
 
 import { addFileToJob, createNewJobItem } from "./service/dynamodbService"
+import {
+  extensionToContentType,
+  File,
+  FileStatus,
+  getFileType,
+} from "./service/jobModel"
 import { getUploadedFilesInJob, getUploadURL } from "./service/S3Client"
 import { modelTiggerSqsEvent, sendMessage } from "./service/sqsClient"
 
 const seenomalySqsURL = process.env.SEENORMALY_URL as string;
+const owlEyeSqsURL = process.env.OWLEUE_URL as string;
 const videoExtension = ["mp4"];
 const imageExtension = ["jpg", "jpeg"];
 
 export interface FileUploadResponseBody {
-	uploadUrl?: string;
+	uploadUrls: {[key:string]:string};
 	jobID: string;
 }
 
 export const createNewJob = async (eventBody: string): Promise<FileUploadResponseBody> => {
 	// 0. parse event body
 	const parsedBody = JSON.parse(eventBody);
-	const jobID = parsedBody["jobID"];
+	const recievedJobID = parsedBody["jobID"];
 	const uploadDone = parsedBody["uploadDone"] === "true";
 	// 3. trigger
 	if (uploadDone) {
-		if (!jobID) {
+		if (!recievedJobID) {
 			throw Error("No jobID provided");
 		}
-		await sqsTriggerModels(jobID);
-		return { jobID } as FileUploadResponseBody;
+		await sqsTriggerModels(recievedJobID);
+		return { jobID: recievedJobID } as FileUploadResponseBody;
 	}
 	const email = parsedBody["email"];
-	const fullFileName = parsedBody["fileName"];
-	const [fileName, fileExtension] = fullFileName.split(".");
-
-	const id = jobID ? jobID : uuidv4();
-	console.log("Running uploadHandler");
-	// 1. upload files to S3
-	const randomfileName = Math.round(Math.random() * 10000000);
-	// append random number to resolve naming conflict, originalName#randomNumber.extension
-	const fileKey = `${id}/${fileName}#${randomfileName}.${fileExtension}`;
-	const uploadUrl = await getUploadURL(fileKey, fileExtension);
+	const fileNames = parsedBody["fileNames"];
+	const jobID = recievedJobID ? recievedJobID : uuidv4();
 	const returnBody: FileUploadResponseBody = {
-		uploadUrl,
-		jobID: id,
+		jobID,
+		uploadUrls: {}
 	};
-
 	// 2. save job to DB
-	if (!jobID) {
-		await createNewJobItem(id, email);
+	if (!recievedJobID) {
+		await createNewJobItem(jobID, email);
 	}
-	await addFileToJob(id, fileKey, email);
+	for (const fileName of fileNames){
+		const file = initFile(fileName, jobID)
+		// 1. generate preSigned Url for files to S3
+		const uploadUrl = await getUploadURL(file.s3Key, file.contentType);
+		// 2. save job to DB
+		await addFileToJob(jobID, file);
+		returnBody.uploadUrls[fileName] = uploadUrl
+
+	}
+
+	console.log("Running uploadHandler");
+	
 	return returnBody;
 };
 
 const sqsTriggerModels = async (jobID: string) => {
 	const uploadedFiles = await getUploadedFilesInJob(jobID);
-	for (const i=0;i < uploadedFiles.length; i++) {
-		const fileKey = uploadedFiles[i]
+	for (let i = 0; i < uploadedFiles.length; i++) {
+		const fileKey = uploadedFiles[i];
```

Comment:
It would be better to use pop() as you've done on line 99, even though the name here should contain only a single period.

Reply:
updated